File: frontend/app/models/comment.js
import DS from "ember-data";

export default DS.Model.extend({
  text: DS.attr('string'),
  empathy_level: DS.attr('number'),
  updated_at: DS.attr('date'),
  created_at: DS.attr('date'),

  // Relationships
  inner_comments: DS.hasMany('inner_comment', {
    async: true
  }),
  user: DS.belongsTo('user', {
    async: true
  }),
  post: DS.belongsTo('post', {
    async: true
  })
});

File: app/models/comment.rb
# Comment Model
class Comment < ActiveRecord::Base
  before_destroy :check_for_inner_comments
  
  # Validations
  validates_presence_of :empathy_level, :text, :user_id, :post_id
  
  # Attributes Length Validations
  validates :text, length: {minimum: 25}
  validates :text, length: {maximum: 1000}

  # Relationships
  has_many :inner_comments
  belongs_to :post, :counter_cache => true
  belongs_to :user, :counter_cache => true
  
  # Make sure that there are no Inner Comments associated before destruction.
  def check_for_inner_comments
    if self.inner_comments.any?
      errors[:base] << "cannot delete Comment when there are inner Comments associated!"
      return false
    end
  end
end
